packages/visualizations/src/components/MapPoi/Map.ts

```diff
@@ -135,13 +141,36 @@ export default class MapPOI {
 
     private bindedResetLeftPaddingPopup = this.resetLeftPaddingPopup.bind(this);
 
+    /** Update popup display between tooltip, sidebar and modal mode */
+    private updatePopupDisplay(display: PopupDisplayTypes) {
+        // Navigation controls could have been removed if the previous popup display was 'modal'.
+        this.queue((map) => map.addControl(this.navigationControl));
+
+        // Remove all popup display classname modifier before adding the current modifier.
+        Object.keys(POPUP_DISPLAY_CLASSNAME_MODIFIER).forEach((d) => {
+            this.popup.removeClassName(POPUP_DISPLAY_CLASSNAME_MODIFIER[d as PopupDisplayTypes]);
+        });
+        this.popup.addClassName(POPUP_DISPLAY_CLASSNAME_MODIFIER[display]);
```

Comment:
I have the feeling here that we should know what popup is present and we don't need to really do everytime a loop, isn't it passed through options ?

Reply:
In https://github.com/opendatasoft/ods-dataviz-sdk/pull/205/commits/7635413b7ebebc37d6388109f1b126cb5b1426d7, add a variable: `activePopupDisplay` to store the current popup display